beta-src/src/utils/map/getOrdersMeta.ts

```diff
@@ -38,7 +39,14 @@ export default function getOrdersMeta(data, phase): Props {
 
       currentOrders.forEach((o) => {
         const { id, unitID, type, toTerrID, fromTerrID } = o;
-        const orderUnit = newBoard.findUnitByID(unitID);
+        let orderUnit = newBoard.findUnitByID(unitID);
+        if (!orderUnit?.Territory && phase === "Retreats") {
+          orderUnit = new UnitClass({
+            ...units[unitID],
+            Territory: territories[units[unitID].terrID],
+          });
+          newBoard.units.push(orderUnit);
+        }
```

Comment:
This seems like a hacky way to set the unit territory. orderUnit has a function called `setTerritory`, you should use that one.